src/superio/nuvoton/nct5104d/superio.c

```diff
@@ -106,6 +107,64 @@ static void route_pins_to_uart(struct device *dev, bool to_uart)
 	pnp_write_config(dev, 0x1c, reg);
 }
 
+static void enable_gpio_io_port(struct device *dev)
+{
+	u8 reg, reg1;
+	u16 io_base_address;
+	u8 uartc_enabled, uartd_enabled;
+
+	// Fix devictree 'enable' bit
+	// Clear LDN 8 CR30.0 bit
+	pnp_write_config(dev, 0x07, NCT5104D_GPIO_WDT);
+	reg = pnp_read_config(dev, 0x30);
+
+	pnp_write_config(dev, 0x30, reg & 0xFE);
+
+	// Check if UARTC and UARTD are both enabled
+	// If they are, don't activate GPIO Address Mode
+	// In any other case - activate GPIO Address Mode
+
+	// select logical device 10
+	pnp_write_config(dev, 0x07, NCT5104D_SP3);
+
+	reg = pnp_read_config(dev, 0x30);
+	uartc_enabled = reg & 0x01;
```

Comment:
@piotr-kleins I would like the static config (i.e. the dev structure) to be checked rather than actual LDN configuration, because it UARTs may be configured sooner or later that this function is executed. You don't know that.

Reply:
@miczyg1 That initialization runs rather late. Logs from device booting confirm that this is last time, when PCI PNP devices are initialized. So I assume, that those registers contains latest UARTC/D state.

Reply:
@piotr-kleins however it is still an assumption. What if someone defines WDT LDN as first device in the device tree SuperIO definition? Then UARTs would get initialized after WDT and it would cause errors